# HG changeset patch
# User Thorsten Meyer <thorsten.meyier@gmx.de>
# Date 1225836230 -3600
#      Tue Nov 04 23:03:50 2008 +0100
# Node ID 33337f1aca75e9d731455fd11dd563c191e25638
# Parent  4d78baf20dedb7b50ac25b554434bc6c28461d55
fix bug in cell string handling of unpack function

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -31,16 +31,21 @@ 2008-11-10  John W. Eaton  <jwe@octave.o
 
 	* polynomial/spline.m: Delete debugging statements.  From
 	Sebastian Schöps <sebastian@schoeps.org>.
 
 2008-11-05  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
         * plot/subplot.m: fix texi bug
 
+2008-11-04  Thorsten Meyer <thorsten.meyier@gmx.de>
+
+        * miscellaneous/unpack.m: return directly after recursive handling
+          of cell-strings
+          
 2008-09-28  Jaroslav Hajek <highegg@gmail.com>
 
 	* optimization/__fdjac__.m: New function file.
 	* optimization/__dogleg__.m: New function file.
 	* optimization/fsolve.m: New function file.
 	* optimization/Makefile.in: Include the new sources.
 	
 2008-09-28  Jaroslav Hajek <highegg@gmail.com>
diff --git a/scripts/miscellaneous/unpack.m b/scripts/miscellaneous/unpack.m
--- a/scripts/miscellaneous/unpack.m
+++ b/scripts/miscellaneous/unpack.m
@@ -92,16 +92,22 @@ function filelist = unpack (file, direct
 
   elseif (iscellstr (file))
     files = {};
     for i = 1:numel (file)
       tmpfiles = unpack (file{i}, directory);
       files = {files{:} tmpfiles{:}};
     endfor
 
+    ## Return output if requested.
+    if (nargout > 0)
+      filelist = files;
+    endif
+
+    return
   else
     error ("unpack: invalid input file class, %s", class(file));
   endif
 
   ## Instructions on what to do for any extension.
   ##
   ## The field names are the file extension without periods.
   ## The first cell is what is executed to unpack an archive verbosely.
